tests/unit_tests/huggingface_pipelines/text.py

```diff
@@ -0,0 +1,180 @@
+from typing import Any, Dict
+from unittest.mock import Mock, patch
+
+import numpy as np
+import pytest
+
+from huggingface_pipelines.text import (
+    EmbeddingToTextPipelineConfig,
+    EmbeddingToTextPipelineFactory,
+    HFEmbeddingToTextPipeline,
+    HFTextToEmbeddingPipeline,
+    TextToEmbeddingPipelineConfig,
+    TextToEmbeddingPipelineFactory,
+)
+
+
+class MockTextToEmbeddingModelPipeline:
+    def __init__(self, encoder, tokenizer, device):
+        pass
+
+    def predict(self, texts, source_lang, batch_size, max_seq_len):
+        return [np.array([0.1, 0.2, 0.3]) for _ in texts]
+
+
+class MockEmbeddingToTextModelPipeline:
+    def __init__(self, decoder, tokenizer, device):
+        pass
+
+    def predict(self, embeddings, target_lang, batch_size):
+        return ["decoded_text" for _ in range(len(embeddings))]
+
+
+@pytest.fixture
+def text_to_embedding_config():
+    return TextToEmbeddingPipelineConfig(
+        encoder_model="test_encoder",
+        columns=["text"],
+        output_column_suffix="embedding",
+        batch_size=2,
+        device="cpu",
+        source_lang="eng_Latn",
+        output_path="test",
+    )
+
+
+@pytest.fixture
+def embedding_to_text_config():
+    return EmbeddingToTextPipelineConfig(
+        decoder_model="test_decoder",
+        columns=["embedding"],
+        output_column_suffix="text",
+        batch_size=2,
+        device="cpu",
+        target_lang="eng_Latn",
+        output_path="test",
+    )
+
+
+@pytest.fixture
+def mock_text_to_embedding_model():
+    with patch(
+        "huggingface_pipelines.text.TextToEmbeddingModelPipeline",
+        MockTextToEmbeddingModelPipeline,
+    ):
+        yield
+
+
+@pytest.fixture
+def mock_embedding_to_text_model():
+    with patch(
+        "huggingface_pipelines.text.EmbeddingToTextModelPipeline",
+        MockEmbeddingToTextModelPipeline,
+    ):
+        yield
+
+
+def test_text_to_embedding_pipeline_initialization(
+    text_to_embedding_config, mock_text_to_embedding_model
+):
+    pipeline = HFTextToEmbeddingPipeline(text_to_embedding_config)
+    assert pipeline.config == text_to_embedding_config
+    assert isinstance(pipeline.t2vec_model, MockTextToEmbeddingModelPipeline)
+
+
+def test_embedding_to_text_pipeline_initialization(
+    embedding_to_text_config, mock_embedding_to_text_model
+):
+    pipeline = HFEmbeddingToTextPipeline(embedding_to_text_config)
+    assert pipeline.config == embedding_to_text_config
+    assert isinstance(pipeline.t2t_model, MockEmbeddingToTextModelPipeline)
+
+
+def test_text_to_embedding_process_batch(
+    text_to_embedding_config, mock_text_to_embedding_model
+):
+    pipeline = HFTextToEmbeddingPipeline(text_to_embedding_config)
+    batch = {"text": [["Hello", "World"], ["Test", "Sentence"]]}
```

Comment:
let's deal with the variable length lists (and maybe use real sentences and not words which may be confusing)

Reply:
let's add more complicated example here (like with len = 1, 0, 2, 4 )

Reply:
DOne.